crates/flux-syntax/src/parser/mod.rs

```diff
@@ -389,7 +432,7 @@ fn mut_as_strg(inputs: Vec<FnInput>, ensures: &[Ensures]) -> ParseResult<Vec<FnI
 ///             ⟨-> ⟨ty⟩⟩?
 ///             ⟨requires⟩ ⟨ensures⟩ ⟨where⟩
 /// ```
-pub(crate) fn parse_fn_sig(cx: &mut ParseCtxt) -> ParseResult<FnSig> {
+pub(crate) fn parse_fn_sig(cx: &mut ParseCtxt, eof: bool) -> ParseResult<FnSig> {
```

Comment:
I would use an arbitrary `end` token and pass `token::Semi` for detached specs, making it more consistent with Rust's syntax.

```suggestion
pub(crate) fn parse_fn_sig<T: PeekExpected>(cx: &mut ParseCtxt, end: T) -> ParseResult<FnSig> {
```

Reply:
done.